examples/console/main.go

```diff
@@ -325,197 +323,219 @@ func run() error {
-			amtRange = amt
-		}
-		for i := 0; i < x; i++ {
-			memo := "tx-" + strconv.Itoa(i)
-			for {
-				amt := amt - (int64(i) % amtRange)
-				_, err = agent.PaymentWithMemo(amt, memo)
-				if err != nil {
-					continue
+
+	shell := ishell.New()
+	shell.AddCmd(&ishell.Cmd{
+		Name: "listen",
+		Help: "listen [addr]:<port> - listen for a peer to connect",
+		Func: func(c *ishell.Context) {
+			err := agent.ServeTCP(c.Args[0])
+			if err != nil {
+				c.Err(err)
+			}
+		},
+	})
+	shell.AddCmd(&ishell.Cmd{
+		Name: "connect",
+		Help: "connect <addr>:<port> - connect to a peer",
+		Func: func(c *ishell.Context) {
+			c.Err(agent.ConnectTCP(c.Args[0]))
+		},
+	})
+	shell.AddCmd(&ishell.Cmd{
+		Name: "open",
+		Help: "open - open a channel with optional asset",
+		Func: func(c *ishell.Context) {
+
+			assetCode := ""
+			if len(c.Args) >= 1 {
+				assetCode = c.Args[0]
+			}
+			asset := state.NativeAsset
+			if assetCode != "" && assetCode != "native" {
+				asset = state.Asset(assetCode + ":" + signer.Address())
+			}
+			c.Err(agent.Open(asset))
+		},
+	})
+	shell.AddCmd(&ishell.Cmd{
+		Name: "deposit",
+		Help: "deposit <amount> - deposit asset into escrow account",
+		Func: func(c *ishell.Context) {
+			depositAmountStr := c.Args[0]
+			destination := escrowAccount
+			if len(c.Args) >= 3 && c.Args[1] == "other" {
+				destination = otherEscrowAccount
+			}
+			account, err := horizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: account.Address()})
+			if err != nil {
+				c.Err(fmt.Errorf("getting state of local escrow account: %w", err))
+			}
+			tx, err := txnbuild.NewTransaction(txnbuild.TransactionParams{
+				SourceAccount:        &account,
+				IncrementSequenceNum: true,
+				BaseFee:              txnbuild.MinBaseFee,
+				Timebounds:           txnbuild.NewTimeout(300),
+				Operations: []txnbuild.Operation{
+					&txnbuild.Payment{Destination: destination.Address(), Asset: asset.Asset(), Amount: depositAmountStr},
+				},
+			})
+			if err != nil {
+				c.Err(fmt.Errorf("building deposit payment tx: %w", err))
+			}
+			tx, err = tx.Sign(networkPassphrase, signer)
+			if err != nil {
+				c.Err(fmt.Errorf("signing deposit payment tx: %w", err))
+			}
+			_, err = horizonClient.SubmitTransaction(tx)
+			if err != nil {
+				c.Err(fmt.Errorf("submitting deposit payment tx: %w", err))
+			}
+		},
+	})
+	shell.AddCmd(&ishell.Cmd{
+		Name: "pay",
+		Help: "pay <amount> - pay amount of asset to peer",
+		Func: func(c *ishell.Context) {
+			amt, err := amount.ParseInt64(c.Args[0])
+			if err != nil {
+				c.Err(err)
+			}
+			stats.Reset()
+			_, err = agent.Payment(amt)
+			c.Err(err)
+		},
+	})
+	shell.AddCmd(&ishell.Cmd{
+		Name: "payx",
+		Help: "payx <amount> <number of times> - pay an amount a certain number of times",
+		Func: func(c *ishell.Context) {
+			fmt.Fprintf(os.Stdout, "sending %s payment %s times\n", c.Args[0], c.Args[1])
+			amt, err := amount.ParseInt64(c.Args[0])
+			if err != nil {
+				c.Err(err)
```

Comment:
Do we need to return at all these points too?

Reply:
yup good catch. c.Err informs shell there's an error but doesn't end the func. Added here: https://github.com/stellar/starlight/pull/403/commits/00b822be0358d759ee31f8450aa7a4504661c71c